keycloak-event-emitter/src/main/java/io/cloudtrust/keycloak/kafkaeventemitter/SerialisationUtils.java

```diff
@@ -0,0 +1,251 @@
+package io.cloudtrust.keycloak.kafkaeventemitter;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.flatbuffers.FlatBufferBuilder;
+import io.cloudtrust.keycloak.customevent.ExtendedAdminEvent;
+import io.cloudtrust.keycloak.customevent.IdentifiedEvent;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Utility class to serialize Event and AdminEvent in Flatbuffer or JSON format.
+ * In order to allow idempotence, a unique ID is added to the entity.
+ */
+class SerialisationUtils {
+    private static final ThreadLocal<ObjectMapper> objectMapperProvider = ThreadLocal.withInitial(ObjectMapper::new);
+
+    private static final int FLATBUFFER_INIT_SIZE = 1024;
+
+    static String toJson(Object obj) throws JsonProcessingException {
+        return objectMapperProvider.get().writeValueAsString(obj);
+    }
+
+    static ByteBuffer toFlat(IdentifiedEvent event) {
+        FlatBufferBuilder builder = new FlatBufferBuilder(FLATBUFFER_INIT_SIZE);
+
+        // uid
+        long uid = event.getUid();
+
+        // Time
+        long time = event.getTime();
+
+        // Type
+        byte type;
+        // size of the list minus 1 because we add UNKNOWN event type in flatbuffers list
+        int eventTypeSize = flatbuffers.events.EventType.names.length - 1;
+
+        if (event.getType().ordinal() < eventTypeSize) {
+            type = (byte) event.getType().ordinal();
+        } else {
+            // EventType returned by the Event is unknown by flatbuffers
+            type = flatbuffers.events.EventType.UNKNOWN;
+        }
+
+        // RealmId
+        int realmId = 0;
```

Comment:
Can't you factorize?
`int realmId = this.getValue(event.getRealmId(), builder)`

Reply:
I did not change anything to serialization